src/Components/Web.JS/src/Boot.WebAssembly.ts

```diff
@@ -124,13 +126,22 @@ async function boot(options?: Partial<WebAssemblyStartOptions>): Promise<void> {
   const bootConfigResult: BootConfigResult = await bootConfigPromise;
   const jsInitializer = await fetchAndInvokeInitializers(bootConfigResult.bootConfig, candidateOptions);
 
+  const progressService = new WebAssemblyProgressService();
+
+  // Only instantiate the WebAssemblyProgressReporter if the loading element exists in the DOM
+  if (options?.setProgress) {
+    progressService.attach(options?.setProgress);
+  } else if (document.getElementById('blazor-default-loading')) {
+    progressService.attach(WebAssemblyProgressReporter.setProgress);
+  }
```

Comment:
Would it work if we inject this with a JS initializer within the template? That way, on the initializer you would set `options.setProgress = WebAssemblyProgressReporter.setProgress` and it will be picked up automatically.

I am not sure we should be harcoding element ids on this file, I would rather us avoid it.

JS initializers for reference https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-6.0#javascript-initializers

Reply:
> Would it work if we inject this with a JS initializer within the template? That way, on the initializer you would set `options.setProgress = WebAssemblyProgressReporter.setProgress` and it will be picked up automatically.

Yep that would work. Alternatively, what I had in mind was:

```js
Blazor.start({
    setProgress: (foo, bar) => { ... }
});
```

in line with how we do [`configureSignalR`](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-7.0#configure-signalr-client-logging-blazor-server).

Is the concern primarily:

> I am not sure we should be harcoding element ids on this file, I would rather us avoid it.

If so, and depending on where we land on the above discussion, one option would be to just have:

```js
WebAssemblyProgressReporter.init(progressService, options)
```

in this file, and then the `init` function for `WebAssemblyProgressReporter` would be:

```js
function init(WebAssemblyProgressService progressService, BlazorWebAssemblyBootOptions options) {
  if (!options?.setProgress && document.getElementById('blazor-default-loading')) {
    progressService.attach(WebAssemblyProgressReporter.setProgress);
  }
}
```

and thus we don't have the hardcoded element in `Boot.WebAssembly`. If however we end up moving `WebAssemblyProgressReporter` to the template (discussion above), then this point becomes moot.